server/xpub-controller/helpers/files.js

```diff
@@ -39,13 +39,16 @@ class FilesHelper {
     pubsub,
     ON_UPLOAD_PROGRESS,
     startedTime,
-    predictedTime,
+    predictor,
     manuscriptId,
   ) {
-    if (manuscriptId.length === 36 && predictedTime > 0) {
+    if (manuscriptId.length === 36 && predictor.getPredictedTime() > 0) {
       return () => {
         const elapsed = Date.now() - startedTime
-        let progress = parseInt((100 * elapsed) / 1000 / predictedTime, 10)
+        let progress = parseInt(
+          (100 * elapsed) / 1000 / predictor.getPredictedTime(),
```

Comment:
Maybe you could store `predictor.getPredictedTime()` as a variable so we don't have to do the calculation process twice?

Reply:
Yeah - great idea ... I was just using `%s/this/that/g`